File: content/07-trees/max_depth_binary_tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def maxDepth(root: TreeNode) -> int:
    if not root:
        return 0
    return 1 + max(maxDepth(root.left), maxDepth(root.right))

if __name__ == "__main__":
    r1 = TreeNode(3, TreeNode(9), TreeNode(20, TreeNode(15), TreeNode(7)))
    assert maxDepth(r1) == 3, "Test 1 Failed"
    r2 = TreeNode(1, None, TreeNode(2))
    assert maxDepth(r2) == 2, "Test 2 Failed"
    print("All tests passed!")

File: content/07-trees/max_depth_binary_tree.js
class TreeNode {
    constructor(val = 0, left = null, right = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

function maxDepth(root) {
    if (!root) return 0;
    return 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
}

const assert = require('assert');
const r1 = new TreeNode(3, new TreeNode(9), new TreeNode(20, new TreeNode(15), new TreeNode(7)));
assert.strictEqual(maxDepth(r1), 3);
const r2 = new TreeNode(1, null, new TreeNode(2));
assert.strictEqual(maxDepth(r2), 2);
console.log("All tests passed!");
